font-codegen/src/record.rs

```diff
@@ -217,6 +217,14 @@ pub(crate) fn generate_compile_impl(
         }
     };
 
+    let maybe_table_type = attrs.tag.is_some().then(|| {
+        quote! {
+            fn type_(&self) -> TableType {
```

Comment:
Just a nit, but the `type_` symbol is pretty unfortunate considering how pervasive it is in the code. Maybe just call it `table_type`?